alert-city/client
# Feat: Internationalization for English Chinese

## TL;DR
- Added language files in messages folder in root directory
- Added dark logo
- Removed All directories in the app folder to [locale] under /app
- Set i18n folder includes request.ts and routing.ts
- Added Preferences.tsx and PreferencesPage.tsx to display Preferences setting
- Completed the internationalization for all components
- New set a page.tsx into app/[locale]/[...rest] to support not found page under next-intl

## Type of Change


## How Has This Been Tested?
1. Login
2. Open the preference setting page in the avatar at the top bar right side
3. Select a new language, see the effect
4. Keep staying login, and change the url to any one doesn't exist, you see the effect
5. Logout,  change the url to any one doesn't exist, you see the effect

## Screenshots
<img width="1718" alt="image" src="https://github.com/user-attachments/assets/a144661a-f12d-43d6-8ba7-39cdb0c80438">

<img width="1718" alt="image" src="https://github.com/user-attachments/assets/2e3b278d-f381-4d89-9833-6b32789f32aa">

<img width="1718" alt="image" src="https://github.com/user-attachments/assets/a08f182a-4792-46e8-8f79-0a9636f053ed">

## Checklist


## Files for Review

1. en.json & zh-cn.json
2. /app/[locale]/[...rest]page.tsx
3. All files in /modules/preference
4. switchLanguage.ts in /utils

## Resolve 
Resolve CP-18
